File: python/helloworld/helloworld.py
#!/usr/bin/env python


import sys

import gi
gi.require_version("Gtk", "3.0")
from gi.repository import Gtk


class MyWindow(Gtk.Window):

    def __init__(self):
        Gtk.Window.__init__(self, title="Hello World")
        Gtk.Window.set_default_size(self, 640, 480)

        self.box = Gtk.Box(spacing=6)
        self.add(self.box)

        self.button = Gtk.Button(label="Hello World")
        self.button.set_halign(Gtk.Align.CENTER)
        self.button.set_valign(Gtk.Align.CENTER)
        self.button.connect("clicked", self.on_button_clicked)
        self.box.pack_start(self.button, True, True, 0)

    def on_button_clicked(self, widget):
        sys.exit()


if __name__ == "__main__":

    win = MyWindow()
    win.connect("destroy", Gtk.main_quit)
    win.show_all()

    Gtk.main()

File: julia/helloworld/helloworld.jl
#!/usr/bin/env julia


using Gtk


function on_button_clicked(w)
  exit()
end


win = GtkWindow("Hello World", 300, 300)


button = GtkButton("Hello World!")
push!(win, button)

signal_connect(on_button_clicked, button, "clicked")

showall(win)


signal_connect(win, :destroy) do widget
    Gtk.gtk_quit()
end

Gtk.gtk_main()
